src/controllers/UserController.ts

```diff
@@ -0,0 +1,70 @@
+/**
+ * Italia PagoPA Proxy
+ * Cittadinanza Digitale PagoPA services
+ *
+ */
+
+import * as Mocked from "../MockedData";
+
+import { none, Option, some } from "fp-ts/lib/Option";
+
+import { CreditCard } from "../types/CreditCard";
+import { PaymentMethod, PaymentMethodType } from "../types/PaymentMethod";
+import { ITransaction } from "../types/Transaction";
+import { IUser } from "../types/User";
+
+/**
+ * User Controller
+ */
+export class UserController {
+    constructor(public user: IUser) { }
+
+    public getWallet(): ReadonlyArray<PaymentMethod> {
+        return Mocked.wallet;
+    }
+
+    public getCreditCards(): ReadonlyArray<CreditCard> {
+        return (
+            Mocked.wallet
```

Comment:
getCreditCards must take the wallet as an argument. make functions "pure", limit the use of global variables.

Reply:
While I agree with you and the "working" version will be pure,this version is using mocked data that is fetched from a global variable. Passing it as an argument will alter the method signature, so I decided against it